src/handlers/guild.ts

```diff
@@ -622,3 +622,100 @@ export function getGuild(guildID: string, counts = true) {
     { with_counts: counts },
   ) as Promise<UpdateGuildPayload>;
 }
+
+/**
+ * Returns the guild template if it exists
+ * @param guildID The ID of the guild
+ * @param code The code of the template to get
+ */
+export function getGuildTemplate(guildID: string, code: string) {
+  const endpoint = `${endpoints.GUILD_TEMPLATES(guildID)}/${code}`;
+  return RequestManager.get(endpoint);
+}
+
+/**
+ * Create a new guild based on a template
+ * NOTE: This endpoint can be used only by bots in less than 10 guilds.
+ * @param code The code of the template to create a guild from
+ */
+export function createGuildFromTemplate(
+  code: string,
+  data: CreateGuildFromTemplate,
+) {
+  return RequestManager.post(endpoints.GUILD_TEMPLATE(code), data);
```

Comment:
Should this return a guild object?

Reply:
We  should also probably handle the urlTobase64 function here for the icon

Reply:
We should also have a check to handle and throw ane rror if the name is tooo long. or too short.